### PR TITLE
User management dashboard

### PR DESCRIPTION
**Pull request by goemen**
Using material ui datatable to display the agency user list, and user account enable/disable actions

### CODE REVIEW

File: backend/api/Areas/Keycloak/Mapping/User/UserMap.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public void Register(TypeAdapterConfig config)
                 .Map(dest => dest.Roles, src => src.Roles)
                 .Inherits<Entity.BaseEntity, Model.Update.BaseModel>();
 
-            config.NewConfig<Model.Update.UserModel, Entity.User>()
```
Comment (Fosol) [inline backend/api/Areas/Keycloak/Mapping/User/UserMap.cs:69]: Are you absolutely certain that this mapping is no longer required anywhere?

File: backend/tests/unit/api/Controllers/Keycloak/UserControllerTest.cs [hunk 2]
```diff
@@ -121,16 +122,16 @@ public async void UpdateUserAsync_Success()
             var service = helper.GetService<Mock<IPimsKeycloakService>>();
             var user = new Entity.User(Guid.NewGuid(), "test", "test@test.com");
             service.Setup(m => m.UpdateUserAsync(It.IsAny<Entity.User>())).Returns(Task.FromResult(user));
-            var model = mapper.Map<Model.Update.UserModel>(user);
```
Comment (Fosol) [inline backend/tests/unit/api/Controllers/Keycloak/UserControllerTest.cs:124]: If this model is no longer required it would be good to delete it.

File: frontend/src/interfaces/pagedItems.tsx [hunk 3]
```diff
@@ -5,5 +5,5 @@ export interface IPagedItems {
   page: number;
   quantity: number;
   total: number;
-  items: [];
+  items: any[];
```
Comment (Fosol) [inline frontend/src/interfaces/pagedItems.tsx:8]: Make the `IPageItems` generic so that items is strongly typed.

File: frontend/src/interfaces/user.tsx [hunk 4]
```diff
@@ -7,4 +7,8 @@ export interface IUser {
   middleName?: string;
   lastName?: string;
   email?: string;
+  username?: string;
+  roles?: any[];
```
Comment (Fosol) [inline frontend/src/interfaces/user.tsx:11]: Make roles and agencies strongly typed.

File: frontend/src/pages/admin/users/interfaces/IUserRecord.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+export enum AccountActive {
+  YES = 'Yes',
+  NO = 'No',
+}
+
+export interface IUserRecord {
```
Comment (Fosol) [inline frontend/src/pages/admin/users/interfaces/IUserRecord.tsx:6]: Should most of these properties be nullable?

File: backend/api/Areas/Keycloak/Mapping/User/UserMap.cs (hunk 1 above)
Comment (goemen) [reply backend/api/Areas/Keycloak/Mapping/User/UserMap.cs:69]: I am certain

File: backend/tests/unit/api/Controllers/Keycloak/UserControllerTest.cs (hunk 2 above)
Comment (goemen) [reply backend/tests/unit/api/Controllers/Keycloak/UserControllerTest.cs:124]: Agreed

File: frontend/src/interfaces/pagedItems.tsx (hunk 3 above)
Comment (goemen) [reply frontend/src/interfaces/pagedItems.tsx:8]: Good idea

File: frontend/src/pages/admin/users/interfaces/IUserRecord.tsx (hunk 5 above)
Comment (goemen) [reply frontend/src/pages/admin/users/interfaces/IUserRecord.tsx:6]: This interface is populated from the IUser interface and the values can be undefined/optional

File: frontend/src/pages/admin/users/ManageUsers.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    state => (state.network as any)[actionTypes.GET_USERS] as IGenericNetworkAction,
+  );
+  useEffect(() => {
+    dispatch(getUsersAction(toApiPaginateParams(0, MAX_USERS_PER_PAGE)));
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const tableChange: TableStateChangeFn = (type: string, state: MUIDataTableState) => {
+    console.log(type, state);
```
Comment (devinleighsmith) [inline frontend/src/pages/admin/users/ManageUsers.tsx:33]: probably meant to take this out.

File: frontend/src/reducers/usersReducer.tsx [hunk 7]
```diff
@@ -20,6 +20,17 @@ const usersReducer = (state = initialState, action: IStoreUsersAction) => {
           items: [...action.pagedUsers.items],
         },
       };
+    case actionTypes.UPDATE_USER:
+      const items = [
+        ...state.pagedUsers.items.map((u: IUser) => (u.id === action.user.id ? action.user : u)),
```
Comment (devinleighsmith) [inline frontend/src/reducers/usersReducer.tsx:25]: I'm curious, why is this necessary?

Comment (goemen) [reply frontend/src/reducers/usersReducer.tsx:25]: What updating the user account status from the table, on success you also have to update/replace the object in the list
